src/types/date.rs

```diff
@@ -0,0 +1,119 @@
+use std;
+use std::fmt;
+use std::fmt::Debug;
+use std::error::Error;
+use neon_runtime;
+use neon_runtime::raw;
+use context::{Context};
+use context::internal::Env;
+use result::{JsResult, JsResultExt};
+use object::{Object};
+use handle::{Handle, Managed};
+use super::{Value, ValueInternal};
+
+/// A JavaScript Date object.
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct JsDate(raw::Local);
+
+impl Value for JsDate { }
+
+impl Managed for JsDate {
+    fn to_raw(self) -> raw::Local { self.0 }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self { JsDate(h) }
+}
+
+#[derive(PartialEq, PartialOrd, Clone, Debug)]
+pub struct DateError(DateErrorKind);
+
+impl DateError {
+    pub fn kind(&self) -> DateErrorKind {
+        self.0
+    }
+}
+
+impl fmt::Display for DateError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.0.as_str())
```

Comment:
FYI, since there is no additional formatting, this can be written more tersely, although I don't think there is much of an impact on the generated code.

```rust
fmt.write_str(self.0.as_str())
```